### ISSUE TITLE
Softmax Loss

### ISSUE THREAD

**Issue by dgomezurita** [ShuangLI59/person_search#92]
Congratulations for your work.

I would like to test your network using the softmax loss function but I still do not understand how to change your train.prototxt. to train correctly. (I have tried several things), I have relied on the branch v_1 however I do not get the results published in your work.
I would like to ask your help to correctly define the train using softmax.
Thank you very much.

**Comment by Cysu** [comment]
@dgomezurita Not sure what your referred to. Is [this layer](https://github.com/ShuangLI59/person_search/blob/v1/models/psdb/VGG16/train.prototxt#L695-L712) you are looking for?

**Comment by dgomezurita** [comment]
Yes, I have seen that layer of your branch v1, I would like to change the lost OIM function in your master branch by the RandomSamplingSoftmaxLoss layer of your v1 branch. Will it be possible to do that?
Should I remove [this layer of OIM](https://github.com/ShuangLI59/person_search/blob/e36a3d9db5d4b21ff29a9618b4e5f818c8f35300/models/psdb/resnet50/train.prototxt#L3077-L3135) [and this] (https://github.com/ShuangLI59/person_search/blob/e36a3d9db5d4b21ff29a9618b4e5f818c8f35300/models/psdb/resnet50/train.prototxt#L3146-L3159) and  I replace them with RandomSamplingSoftmaxLoss?

**Comment by dgomezurita** [comment]
Yes, I have seen that layer of your branch v1, I would like to change the lost OIM function in your master branch by the RandomSamplingSoftmaxLoss layer of your v1 branch. Will it be possible to do that?
Should I remove [this layer of OIM](https://github.com/ShuangLI59/person_search/blob/e36a3d9db5d4b21ff29a9618b4e5f818c8f35300/models/psdb/resnet50/train.prototxt#L3077-L3135) [and this](https://github.com/ShuangLI59/person_search/blob/e36a3d9db5d4b21ff29a9618b4e5f818c8f35300/models/psdb/resnet50/train.prototxt#L3146-L3159) and  I replace them with RandomSamplingSoftmaxLoss?

**Comment by Cysu** [comment]
If you want to use RandomSamplingSoftmax rather than OIM loss, you can directly use branch v1.

**Comment by dgomezurita** [comment]
Yes, that is correct, but I expressed myself badly, I would like to use the convolutional network resnet-50 of its branch master instead of VGG of branch v1, what is your recommendation to try SoftmaxRS or conventional Softmax?
thank you very much.

**Comment by Cysu** [comment]
We have a baseline of using conventional Softmax loss in the paper. I don't think it can work in this person search framework.

We also have an experiment of subsampling unlabeled identities in the paper. I think SoftmaxRS might work but OIM is a better choice due to its non-parametric nature.

**Comment by dgomezurita** [comment]
thank you!
